packages/core/src/trait.ts

```diff
@@ -5,10 +5,12 @@ import { parseVersion, Version } from './version';
 
 // spec
 
+type TraitMetaData = Metadata<{ hasUnmount: boolean }>;
```

Comment:
`hasUnmount` sounds ad-hoc for this scenario, is there some general word to describe this pattern(execute before render/skip render)?

Reply:
ok, I have changed to `beforeRender`.